docs/dictionary.rst

```diff
@@ -0,0 +1,994 @@
+
+see also :ref:`dict GitHub App registration`  
+
+
+
+.. _dict GitHub App workflow:
+
+GitHub App workflow
+-------------------
+    A :ref:`dict workflow` registered with the Dockstore GitHub App.  
+
+.. note:: This term as we define it here is associated with Dockstore and may have different definitions in other contexts.  
+
+see also :ref:`dict GitHub App registration`  
+
+
+
+.. _dict immutable:
+
+immutable
+---------
+    Unchanging, unable to be modified. Immutability implies that an object cannot be updated.  
+
+
+
+
+.. _dict interoperable:
+
+interoperable
+-------------
+    The ability of data or tools from multiple resources to effectively integrate data, or operate processes, across all systems with a moderate degree of effort.  
+
+
+
+
+.. _dict JSON:
+
+JSON
+----
+pronounced "jason"  
+
+*abbreviation for* JavaScript Object Notation  
+
+    A human-readable file format that originated in JavaScript, but is now used by a variety of applications. Dockstore supports the inclusion of JSON and :ref:`dict YAML` files in entries to provide sample inputs for workflow and tool entries. Some workflow executors, such as :ref:`dict Cromwell`, can use these files to configure their inputs rather than having to manually listing every input when calling the workflow on the command line.  
+
+see also :ref:`dict YAML`  
+
+Further reading: `<https://www.json.org/json-en.html>`_  
+
+
+
+.. _dict Jupyter:
+
+Jupyter
+-------
+pronounced "Jupiter" like the planet  
+
+    A project focused on developing "notebooks" for programming languages, most famously Python due to it starting as a splinter of iPython in the early 2010s. Other languages such as R are also supported. Jupyter notebooks allow for blocks of code to be nestled between markdown text, allowing for easy documentation of the code blocks and reproducibility of analysis.  
+
+Further reading: `<https://jupyter.org/>`_  
+
+
+
+.. _dict kernel:
+
+kernel
+------
+    An operating system's core program that is always loaded in memory, and modulates interactions between software and physical hardware, including but not limited to managing memory access for any program currently in RAM.  
+
+Further reading: `<https://en.wikipedia.org/wiki/Kernel_(operating_system)>`_  
+
+
+
+.. _dict Kids First:
+
+Kids First
+----------
+*abbreviation for* Gabriella Miller Kids First Program  
+
+    An :ref:`dict NIH` program, supported by the NIH Common Fund, relating to the influence of genomics on pediatric health, with a focus on pediatric cancer and structural birth abnormalities (such as cleft palate).  
+
+Further reading: `<https://commonfund.nih.gov/kidsfirst/highlights>`_  
+
+
+
+.. _dict labels:
+
+labels
+------
+    On Dockstore, we use labels to "tag" Dockstore entries with information about them. Workflow or tool developers can add labels to a Dockstore :ref:`dict entry` page that they have edit access to. Labels cannot contain spaces.  
```

Comment:
Instead of "Labels cannot contain spaces", I would mention that labels show up on the search page.